MISRA.md

```diff
@@ -10,28 +10,28 @@ Deviations from the MISRA standard are listed below:
+| Rule 10.5 | Advisory | Numerical casting is done only for the `ucTCPState` member of `xTCP` structure to cast it to `eIPTCPState_t` enum which has been verified to be big enough to store all values in the `ucTCPState`. |
 | Rule 11.8 | Required | `const` identifier is removed in some cases since certain `struct`s are made to be used with modifiable types. |
 | Rule 14.4 | Required | Condition `0` not having an essentially boolean type is used in portable layer of the Kernel for `taskENTER_CRITICAL` and `taskEXIT_CRITICAL`. |
 | Rule 8.6 | Required | Some functions are declared but not defined since they are to be defined by the user application to provide 'hooks' for certain events. |
 | Rule 2.3 | Advisory |  |
-| Rule 20.10 | Advisory | `#` or `##` is used to create some utility functions for casting variables.  |
+| Rule 20.10 | Advisory | For casting pointers of one type to another, FreeRTOS-Plus-TCP uses utility functions which are created using `##` operator to make signature of functions similar and avoid user the pain of defining functions for various types. |
 | Rule 10.4 | Required | Signed and unsigned variables are used together due to the variables acquired from the portable code. |
-| Rule 13.5 | Required | Persistent side effects are required in some function calls (such as to maintain the state of a state-machine etc). |
-| Rule 15.5 | Advisory | `return` statement is restrictively used to optimise the response time of the stack. |
+| Rule 13.5 | Required | In FreeRTOS_TCP_IP.c (in function `prvTCPSendPacket`) persistent side effect of function call as right-hand operators of `||` has been verified to behave as expected. In FreeRTOS_Sockets.c (in function `FreeRTOS_sendto`) the violation noted is false positive. |
+| Rule 15.5 | Advisory | Some FreeRTOS-Plus-TCP functions use multiple `return` statements; These have not been refactored to comply with this rule. |
 | Rule 18.3 | Required | Relational operators are used with members of unions. |
-| Rule 21.6 | Required | `snprintf` is used restrictively to format strings before placing them into buffers. |
-| Rule 10.8 | Required | Unsigned expression is casted in one location to compare with a signed variable.  |
-| Rule 11.1 | Required | `void *` is used to pass variables of various types to a function as parameter. |
-| Rule 14.3 | Required | Controlling expression is not invariant in some cases due to dependence on a `#if`. |
-| Rule 2.2 | Required | There is no dead code, just use of unions which confuses Coverity compiler. |
+| Rule 21.6 | Required | In module FreeRTOS_IP.c (in function `FreeRTOS_strerror_r`) `snprintf` is used to format strings before placing them into buffers. |
+| Rule 10.8 | Required | Unsigned expression in function `prvSendData` (present in FreeRTOS_TCP_IP.c) is cast to signed to compare with a signed variable. |
```

Comment:
This line and line 32 are worded differently than lines 27 and 30.  Make the wording consistent with those lines - mention the module name first and then the function in parentheses.